core/src/main/java/org/infinispan/configuration/cache/Configurations.java

```diff
@@ -66,4 +67,12 @@ public static boolean hasSharedCacheLoaderOrWriter(Configuration configuration)
       }
       return false;
    }
+
+   public static boolean isRecoveryEnabled(Configuration configuration) {
+      TransactionConfiguration txConfig = configuration.transaction();
+      return txConfig.transactionMode() == TransactionMode.TRANSACTIONAL &&  //tx cache
+            !txConfig.useSynchronization() && //xa enlistment
+            txConfig.transactionProtocol() != TransactionProtocol.TOTAL_ORDER && //non TO cache
+            txConfig.recovery().enabled(); //recovery enabled
```

Comment:
well, it cross my mind that we can have a ```Configuration``` instance without validation. You could create a new instance of ```Configuration``` by hand or using ```ConfigurationBuilder.build(false)``` (skips validation)